windows/src/desktop/setup/RunTools.pas

```diff
@@ -610,8 +611,26 @@ function TRunTools.InstallMSI: Boolean;
   if not IsNewerVersionInstalled(FInstallInfo.Version) then // I2560
   begin
     ReinstallMode := 'REBOOTPROMPT=S REBOOT=ReallySuppress'; // I2754 - Auto update is too silent
-    if (FInstalledVersion.Version <> '') and (FInstalledVersion.ProductCode = FInstallerVersion.ProductCode)
-      then ReinstallMode := ReinstallMode + ' REINSTALLMODE=vomus REINSTALL=ALL';
+    if (FInstalledVersion.Version <> '') and (FInstalledVersion.ProductCode = FInstallerVersion.ProductCode) then
+    begin
+      ReinstallMode := ReinstallMode + ' REINSTALLMODE=vomus REINSTALL=ALL';
+    end
+    else
+    begin
+      Status('Removing older versions');
+      // Remove older versions of Keyman now. We'll still get the upgrade desired
+      // because we've backed up the relevant keys for reapplication post-install
```

Comment:
From your PR description, it sounds like this check only needs to be done for upgrading from: Keyman 7, 8, 9, 10.

Maybe add a comment for future maintenance so this list won't need to be continually updated.